subprojects/docs/src/docs/userguide/jvm/jvm_test_suite_plugin.adoc

```diff
@@ -0,0 +1,165 @@
+
+[[sec:jvm_test_suite_examples]]
+[discrete]
+== Configuration Examples
+
+Here are several examples to illustrate the configurability of test suites.
+
+== Declaring an additional test suite
+====
+include::sample[dir="snippets/testing/test-suite-plugin/groovy",files="build.gradle[tags=configure-testing-extension]"]
+include::sample[dir="snippets/testing/test-suite-plugin/kotlin",files="build.gradle.kts[tags=configure-testing-extension]"]
+====
+
+<1> Configure all test suites for this project
+<2> Configure the built-in `test` suite.  This suite is automatically created for backwards compatibility.  By convention, the built-in `test` suite will use JUnit4.
+<3> Declare this test suite uses JUnit Jupiter.  The framework's dependencies are automatically included.  It is not necessary to explicitly name or configure the built-in `test` suite if JUnit4 is desired.
+<4> Define a new suite called `integrationTest`.
+<5> Add a dependency on the production code of the project to the `integrationTest` suite targets.  By convention, only the built-in `test` suite will automatically have a dependency on the production code of the project.
+<6> Configure all targets of this suite.  By convention, test suite targets have no relationship to other `Test` tasks.  This example shows that the slower integration test suite targets should run after all targets of the `test` suite are complete.
+<7> Configure the `check` task to depend on all `integrationTest` targets.  By convention, test suite targets are not associated with the `check` task.
+
+Invoking the `check` task on the above configured build should show output similar to:
+
+[listing]
+----
+include::{snippetsPath}/testing/test-suite-plugin/tests/checkTaskOutput.out[]
+----
+
+Note that all tasks synthesized by the registration of the `integrationTest` suite (`compileIntegrationTestJava`, `processIntegrationTestResources`, `integrationTestClasses` and `integrationTest`) do not run until after the `test` suite completes its tasks.
+
+== Configuring the built-in `test` suite
+====
+include::sample[dir="snippets/testing/test-suite-configure-default-suite/groovy",files="build.gradle[tags=configure-default-suite]"]
+include::sample[dir="snippets/testing/test-suite-configure-default-suite/kotlin",files="build.gradle.kts[tags=configure-default-suite]"]
+====
+
+// TODO test with `./gradlew :docs:docsTest --tests "*test-suite-configure-default-suite*"`
+<1> All targets of the `test` test suite will use the TestNG framework
+<2> Lazily configure all targets of the suite; note the return type of `testTask` is `TaskProvider<Test>`.
+<3> More detailed options of the specific test framework can be configured here.  The `options` will be a subclass of `org.gradle.api.tasks.testing.TestFrameworkOptions`, in this case it is `org.gradle.api.tasks.testing.testng.TestNGOptions`
+
+
+== Configuring dependencies of a suite
```

Comment:
This is another source of confusion, I think.

Could you add a couple of sentences here that tries to avoid the confusion someone may have with:
```
dependencies {
   implementation "foo:bar:1.0"
}
```
and
```
testing {
   suites {
      test {
         dependencies {
            implementation "foo:bar:1.0"
         }
      }
   }
}
```
Not doing the same thing?

I think what we want to say is that the dependencies block within a test suite is already scoped for that test suite. Instead of having to know the global name of the configuration, test suites have a consistent name that you use in this block for declaring implementation, compileOnly and runtimeOnly dependencies.